inference/core/workflows/core_steps/models/foundation/florence2/v1.py

```diff
@@ -0,0 +1,222 @@
+from typing import List, Literal, Optional, Type, TypeVar, Union
+
+from pydantic import ConfigDict, Field, model_validator
+import numpy as np
+import supervision as sv
+from pydantic import ConfigDict, Field
+
+from inference.core.entities.requests.inference import LMMInferenceRequest
+from inference.core.entities.responses.inference import LMMInferenceResponse
+from inference.core.managers.base import ModelManager
+from inference.core.workflows.core_steps.common.entities import StepExecutionMode
+from inference.core.workflows.core_steps.common.utils import (
+    attach_parents_coordinates_to_batch_of_sv_detections,
+    attach_prediction_type_info_to_sv_detections_batch,
+    convert_inference_detections_batch_to_sv_detections,
+    load_core_model,
+)
+from inference.core.workflows.execution_engine.entities.base import (
+    Batch,
+    OutputDefinition,
+    WorkflowImageData,
+)
+from inference.core.workflows.execution_engine.entities.types import (
+    BOOLEAN_KIND,
+    FLOAT_KIND,
+    INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    KEYPOINT_DETECTION_PREDICTION_KIND,
+    OBJECT_DETECTION_PREDICTION_KIND,
+    LANGUAGE_MODEL_OUTPUT_KIND,
+    STRING_KIND,
+    ImageInputField,
+    StepOutputImageSelector,
+    StepOutputSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+T = TypeVar("T")
+K = TypeVar("K")
+
+DETECTIONS_CLASS_NAME_FIELD = "class_name"
+DETECTION_ID_FIELD = "detection_id"
+
+LONG_DESCRIPTION = """
+Run Florence-2, a large multimodal model, on an image.
+
+** Dedicated inference server required (GPU recomended) **
+"""
+
+TaskType = Literal[
+    "<OCR>",
```

Comment:
This is change in favour of clients using UI, probably would decrease code readability, but I would still choose that tradeoff - let's change the literal entries into this style: https://github.com/roboflow/inference/blob/aa5b4b031a97c1ccd32fcbeb2ebdd7c8333a8584/inference/core/workflows/core_steps/models/foundation/google_gemini/v1.py#L76

`ocr-with-region` instead `<OCR_WITH_REGION>` and change it in prompting into `f"<{value.upper().replace('-', '_')}>"`

Reply:
Sorry, this was still draft -- I added legible english descriptions and mapped them to tasks... that ok?